### PR TITLE
Create section for installing from private package repos

### PR DESCRIPTION
**Pull request by stuart23**
The approach is very similar to installing from a GH repo, but in this case we are installing prebuilt packages that are hosted on a private package repository. This was already asked in [ZD 8829](https://astronomer.zendesk.com/agent/tickets/8829) and so I think it would be good to show how this is done. I think we will see more and more people switching towards prepackaged packages because it's a better way of distributing packages - however until Github do it, it might not become that popular (https://github.com/github/roadmap/issues/94)

### CODE REVIEW

**Comment by jwitz** [comment]
@stuart23 Is the technical content here finalized? One we merge the other private repo PR, I'm going to jump in here and use Docusaurus' [tabs](https://docusaurus.io/docs/markdown-features/tabs) feature to condense these setups into a single section, since overall they're pretty similar. Won't get to this until I'm back from vacation on May 2nd though!

**Comment by stuart23** [comment]
@jaketf, @paolaperaza and I were just talking about separating these by tabs - I think it's a great idea! The technical details should be final, but you may want to play with some of the wording.

Happy to close the PR, or leave it here in draft mode until you return. Have a great vacation! :sunglasses::desert_island:

**Comment by stuart23** [comment]
Had another customer request to do this https://astronomer.zendesk.com/agent/tickets/9443

File: astro/develop-project.md [hunk 1]
```diff
@@ -357,7 +359,25 @@ my_project
...
+The image usually used to build (quay.io/astronomer/astro-runtime:4.2.10) contains [ONBUILD](https://docs.docker.com/engine/reference/builder/#onbuild) instructions that will automatically
+install the dependancies from `packages.txt` and `requirements.txt`. For packages to be installed from private sources, an *unpacked* version of the image must be used to allow credentials to
+be injected before the ONBUILD instructions are executed.
+
+<Tabs
+    defaultValue="github"
+    values={[
+        {label: 'Private GitHub Repo', value: 'github'},
+        {label: 'Private PyPi Index', value: 'pypi'},
+        {label: 'Linux', value: 'linux'},
```
Comment (jwitz) [inline astro/develop-project.md:376]:
```suggestion
```

File: astro/develop-project.md [hunk 2]
```diff
@@ -357,7 +359,24 @@ my_project
     └── prod.env
 ```
 
-## Install Python Packages from a Private GitHub Repository
+## Install Python Packages from Private Sources
+
+Python packages can be installed from public and private locations into your image. Packages listed on [PyPI](https://pypi.org/search/) can easily be installed by simply listing them in the
+`requirements.txt`. For packages listed on private PyPI indices, or whos source code is stored in a private git repository, the build process requires some extra steps.
```
Comment (jwitz) [inline astro/develop-project.md:365]:
```suggestion
Python packages can be installed from public and private locations into your image. To install public packages listed on [PyPI](https://pypi.org/search/), follow the steps in [Add Python and OS-level Packages](add-python-and-os-level-packages). To install packages listed on private PyPI indices or a private git-based repository, you need to complete additional configuration in your project.
```

File: astro/develop-project.md [hunk 3]
```diff
@@ -357,7 +359,24 @@ my_project
...
+
+The image usually used to build (quay.io/astronomer/astro-runtime:4.2.10) contains [ONBUILD](https://docs.docker.com/engine/reference/builder/#onbuild) instructions that will automatically
+install the dependancies from `packages.txt` and `requirements.txt`. For packages to be installed from private sources, an *unpacked* version of the image must be used to allow credentials to
+be injected before the ONBUILD instructions are executed.
```
Comment (jwitz) [inline astro/develop-project.md:369]:
```suggestion
Depending on where your private packages are stored, use one of the following setups to install your packages to an Astro project by customizing your Runtime image. 
```

File: astro/develop-project.md [hunk 4]
```diff
@@ -499,4 +518,110 @@ This example assumes that the name of each of your Python packages is identical
...
-Your Astro project can now utilize Python packages from your private GitHub repository. To test your DAGs, you can either [run your project locally](develop-project.md#build-and-run-a-project-locally) or [deploy to Astro](deploy-code.md).
+Your Astro project can now utilize Python packages from your private GitHub repository. To test your DAGs, you can either [run your project locally](develop-project.md#build-and-run-a-project-locally) or [deploy to Astro](deploy-cli.md).
+
+</TabItem>
+
+<TabItem value="pypi">
+
+#### Install Python Packages from a Private PyPI Index
+
+This topic provides instructions for building your Astro project using Python packages from a private PyPI index. In some organizations, python packages will be prebuilt and pushed to a hosted private pip server (such as pypiserver, or Nexus Repository) or managed service (such as PackageCloud or Gitlab). Similar to [installing from a private GitHub Repository](Install Python Packages from a Private GitHub Repository), builds that require packages from a private PyPI index must be done using a multi-stage build.
```
Comment (jwitz) [inline astro/develop-project.md:529]:
```suggestion
This topic provides instructions for building your Astro project using Python packages from a private PyPI index. In some organizations, python packages are prebuilt and pushed to a hosted private pip server (such as pypiserver or Nexus Repository) or managed service (such as PackageCloud or Gitlab). At a high level, this setup requires specifying your private packages in `requirements.txt`, creating a custom Docker image that changes where pip looks for packages, and building your project with this Docker image.
```

File: astro/develop-project.md [hunk 5]
```diff
@@ -499,4 +518,110 @@ This example assumes that the name of each of your Python packages is identical
...
+1. Run the following command to create a new Docker image from your `Dockerfile.build` file, making sure to substitute in the pip repository and associated credentials:
+
+    ```sh
+    DOCKER_BUILDKIT=1 docker build -f Dockerfile.build --progress=plain --build-arg PIP_EXTRA_INDEX_URL=https://${<repo-username>}:${<repo-password>}@<private-pypi-repo-domain-name> -t custom-<airflow-image> .
+    ```
+
+    For example, if you have `quay.io/astronomer/astro-runtime:5.0.0` in your `Dockerfile.build`, this command would be:
+
+    ```sh
+    DOCKER_BUILDKIT=1 docker build -f Dockerfile.build --progress=plain --ssh=github="$HOME/.ssh/<authorized-key>" -t custom-astro-runtime-5.0.0 .
```
Comment (jwitz) [inline astro/develop-project.md:608]: I don't think this example has been updated for the new setup

File: astro/develop-project.md [hunk 6]
```diff
@@ -499,4 +518,110 @@ This example assumes that the name of each of your Python packages is identical
...
+   FROM custom-<airflow-image>
+   ```
+
+   For example, if your base Runtime image was `quay.io/astronomer/astro-runtime:5.0.0`, this line would be:
+
+   ```
+   FROM custom-astro-runtime:5.0.0
+   ```
+
+   Your Astro project can now utilize Python packages from your private GitHub repository. To test your DAGs, you can either [run your project locally](develop-project.md#build-and-run-a-project-locally) or [deploy to Astro](deploy-code.md).
```
Comment (jwitz) [inline astro/develop-project.md:624]:
```suggestion
   Your Astro project can now utilize Python packages from your private PyPi index. To test your DAGs, you can either [run your project locally](develop-project.md#build-and-run-a-project-locally) or [deploy to Astro](deploy-code.md).
```

File: astro/develop-project.md [hunk 7]
```diff
@@ -499,4 +518,110 @@ This example assumes that the name of each of your Python packages is identical
...
+    COPY --from=stage2 /usr/local/lib/python3.9/site-packages /usr/local/lib/python3.9/site-packages
+    COPY . .
+    ```
+
+    In order, these commands:
+
+    - Complete the standard installation of OS-level packages in `packages.txt`.
+    - Add the environment variable `PIP_EXTRA_INDEX_URL` to instruct pip on where to look for non-public packages.
+    - Install public and private Python-level packages from your `requirements.txt` file.
+
```
Comment (jwitz) [inline astro/develop-project.md:595]:
```suggestion
```

**Review by jwitz — CHANGES_REQUESTED** [review]
Nice PR @stuart23 ! I jumped in to the branch to add tab formatting, but kept my other changes as suggestions in case you had thoughts on my wording. Let me know what you think, and I'll do one final pass to make sure the formatting looks OK before merging.

**Comment by stuart23** [comment]
@jwitz checked it in Netlify and it is :kissing_heart::ok_hand:

I was thinking of adding the following extra tidbit at line 586, but do you think it's too in-the-weeds/best-practices?

    :::tip

    It is usually advised not to inject secrets literally as they can be easily uncovered using `docker history` on the image. In this case, a multi-stage build method is being used, and the secret is not in scope for stage3, and will not appear in the history.

    :::

File: astro/develop-project.md (hunk 5 above)
Comment (jwitz) [reply astro/develop-project.md:603]: Pingin on this again

File: software/customize-image.md [hunk 8]
```diff
@@ -391,6 +404,110 @@ This example assumes that the name of each of your Python packages is identical
...
+1. Run the following command to create a new Docker image from your `Dockerfile.build` file, making sure to substitute in the pip repository and associated credentials:
+
+    ```sh
+    DOCKER_BUILDKIT=1 docker build -f Dockerfile.build --progress=plain --build-arg PIP_EXTRA_INDEX_URL=https://${<repo-username>}:${<repo-password>}@<private-pypi-repo-domain-name> -t custom-<airflow-image> .
+    ```
+
+    For example, if you have `quay.io/astronomer/ap-airflow:2.2.5` in your `Dockerfile.build`, this command would be:
+
+    ```sh
+    DOCKER_BUILDKIT=1 docker build -f Dockerfile.build --progress=plain --ssh=github="$HOME/.ssh/<authorized-key>" -t custom-ap-airflow:2.2.5 .
```
Comment (jwitz) [inline software/customize-image.md:490]: Same thing here

**Review by jwitz — APPROVED** [review]
I think this is almost done @stuart23 ! There's just one more reference to GitHub in the PyPi docs that maybe needs to be fixed for both Astro and Software, but other than that this LGTM

The other note you suggested doesn't help users make a decision or take action during this build, so I'd advise against including it. If you want, you could add another inline comment to the Dockerfile that explains why our Stage 3 step is safe, but don't stress it either way

File: software/customize-image.md (hunk 8 above)
Comment (stuart23) [reply software/customize-image.md]: :woozy_face: My bad

**Pull request merged**